BlueprintUI/Sources/AttributedText/AttributedText.swift

```diff
@@ -0,0 +1,158 @@
+import UIKit
+
+/// `AttributedText` allows you to apply strongly-typed attributes to strings (much like the `AttributedString` type
+/// introduced in iOS 15). You can then access the `attributedString` property to get an attributed string with those
+/// attributes applied.
+///
+/// For example:
+///
+/// ```swift
+/// var text = AttributedText(string: "Hello, world")
+/// // Apply a font to the entire range
+/// text.font = .systemFont(ofSize: 20)
+///
+/// // Apply a color to part of the string
+/// let range = text.string.range(of: "world")!
+/// text[range].color = .blue
+///
+/// // Render the attributed text
+/// let label = AttributedLabel(attributedText: text.attributedString)
+/// ```
+///
+@dynamicMemberLookup public struct AttributedText {
+
+    /// The wrapped string, with no attributes.
+    public let string: String
+
+    /// An `NSAttributedString` representation of the attributed text.
+    public var attributedString: NSAttributedString {
+        mutableAttributedString
+    }
+
+    private var mutableAttributedString: NSMutableAttributedString
+
+    /// Create some `AttributedText` from a plain string.
+    public init(string: String) {
+        self.string = string
+        self.mutableAttributedString = NSMutableAttributedString(string: string)
+    }
+
+    /// Create some `AttributedText` from an attributed string. The attributes are preserved.
+    public init(attributedString: NSAttributedString) {
```

Comment:
Should this also just be `string:`? Should both inits have no name for the parameter?

```suggestion
    public init(string: NSAttributedString) {
```

Reply:
I'm pretty sure we want to omit the parameter here based on the [Swift API Design Guidelines](https://www.swift.org/documentation/api-design-guidelines/#type-conversion), but I might be misunderstanding them.

Reply:
Or maybe we are supposed to put the parameter for `String` because it's "narrowing" and not put the parameter for `NSAttributedString` because it's just converting?

Reply:
I'm for that